examples/multiple_simple_compression.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2016-present, Yann Collet, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under both the BSD-style license (found in the
+ * LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ * in the COPYING file in the root directory of this source tree).
+ * You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <stdlib.h>    // malloc, free, exit
+#include <stdio.h>     // fprintf, perror, fopen, etc.
+#include <string.h>    // strlen, strcat, memset, strerror
+#include <errno.h>     // errno
+#include <sys/stat.h>  // stat
+#include <zstd.h>      // presumes zstd library is installed
+#include "utils.h"
+
+typedef struct {
+    void* fBuffer;
+    void* cBuffer;
+    size_t fBufferSize;
+    size_t cBufferSize;
+    ZSTD_CCtx* cctx;
+} resources;
+
+/*
+ * allocate memory for buffers big enough to compress all files
+ * as well as memory for output file name (outFilename)
+ */
+static resources createResources_orDie(int argc, const char** argv, char **outFilename)
```

Comment:
Could `outFileName` be part of `resources`, since it's basically just a buffer to manipulate strings ?

Reply:
Totally reasonable !

For the record, another design pattern could be to create an encapsulating `struct` as return type of `createResources_orDie()`, containing both the current `resources` and `outFilename` buffer, so that they can be easily separated later for `compressFile_orDie()`.

However, that would define 2 different types of `resources`, requiring different names to avoid confusing. 
It's probably overkill for such a simple example.